database/src/migrations/20230817000000_critter_observation_deployment_tables.ts

```diff
@@ -0,0 +1,184 @@
+  COMMENT ON COLUMN observation.update_date              IS 'The datetime the record was updated.';
+  COMMENT ON COLUMN observation.update_user              IS 'The id of the user who updated the record as identified in the system user table.';
+  COMMENT ON COLUMN observation.revision_count           IS 'Revision count used for concurrency control.';
+  COMMENT ON TABLE  observation                          IS 'Information about taxon counts.';
+
+  -- Create audit and journal triggers
+  create trigger audit_observation before insert or update or delete on observation for each row execute procedure tr_audit_trigger();
+  create trigger journal_observation after insert or update or delete on observation for each row execute procedure tr_journal_trigger();
+
+  ----------------------------------------------------------------------------------------
+  -- Create Critter table
+  ----------------------------------------------------------------------------------------
+
+  CREATE TABLE critter(
+    critter_id               integer           GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1),
+    observation_id           integer           NOT NULL,
+    critterbase_critter_id   uuid              NOT NULL,
+    create_date              timestamptz(6)    DEFAULT now() NOT NULL,
+    create_user              integer           NOT NULL,
+    update_date              timestamptz(6),
+    update_user              integer,
+    revision_count           integer           DEFAULT 0 NOT NULL,
+    CONSTRAINT critter_pk PRIMARY KEY (critter_id)
+  );
+
+  COMMENT ON COLUMN critter.critter_id               IS 'System generated surrogate primary key identifier.';
+  COMMENT ON COLUMN critter.observation_id           IS 'The id of the observation';
+  COMMENT ON COLUMN critter.critterbase_critter_id   IS 'The external system id of a Critterbase critter.';
+  COMMENT ON COLUMN critter.create_date              IS 'The datetime the record was created.';
+  COMMENT ON COLUMN critter.create_user              IS 'The id of the user who created the record as identified in the system user table.';
+  COMMENT ON COLUMN critter.update_date              IS 'The datetime the record was updated.';
+  COMMENT ON COLUMN critter.update_user              IS 'The id of the user who updated the record as identified in the system user table.';
+  COMMENT ON COLUMN critter.revision_count           IS 'Revision count used for concurrency control.';
+  COMMENT ON TABLE  critter                          IS 'Information about individual animals associated to specific observations.';
+
+  -- Add foreign key constraint from child table to parent table on observation_id
+  ALTER TABLE critter ADD CONSTRAINT critter_fk1
+    FOREIGN KEY (observation_id)
+    REFERENCES observation(observation_id);
+
+  -- Add unique constraint
+  CREATE UNIQUE INDEX critter_uk1 ON critter(critterbase_critter_id);
```

Comment:
I think we don't want this index to be unique. Over time, multiple surveys/observations could observe the same critter.

Also, we should add an index for `observation_id`. (Another funny thing about postgres, is that foreign keys dont come with indexes automatically like they do in other sql languages, so every time we add a foreign key, we also need to add an index).

Reply:
I think we do want a unique index on `observation_id + critterbase_critter_id` though. Since it wouldn't make sense for the same observation to observe the same animal more than once.